sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/util/CharVarcharCodegenUtils.java

```diff
@@ -26,7 +27,7 @@ private static UTF8String trimTrailingSpaces(
       UTF8String inputStr, int numChars, int limit) {
     int numTailSpacesToTrim = numChars - limit;
     UTF8String trimmed = inputStr.trimTrailingSpaces(numTailSpacesToTrim);
-    if (trimmed.numChars() > limit) {
+    if (trimmed.numChars() > limit && !SQLConf.get().charVarcharAsString()) {
```

Comment:
We don't need this now.

Reply:
then do we need update PartitioningUtils.charTypeWriteSideCheck and PartitioningUtils.varcharTypeWriteSideCheck ?

Reply:
How do we plan to update these two?